aten/src/ATen/native/mkldnn/xpu/Conv.cpp

```diff
@@ -0,0 +1,726 @@
+#include <vector>
+
+#include <detail/oneDNN.h>
+#include <ATen/core/ATen_fwd.h>
+#include <ATen/core/interned_strings.h>
+#include <ATen/ops/full.h>
+#include <ATen/ops/neg.h>
+#include <c10/core/Scalar.h>
+#include <c10/util/Exception.h>
+#include <c10/util/Optional.h>
+#include <ATen/native/utils/ParamUtils.h>
+#include <torch/library.h>
+
+using namespace dnnl;
+using namespace at::native;
+using namespace at::native::xpu::onednn;
+
+namespace at::native {
+namespace xpu {
+namespace impl {
+
+struct ConvParams {
```

Comment:
should we reuse ConvParams in aten/src/ATen/native/Convolution.cpp ?

Reply:
Thanks for suggestion, but this class is not part of API we could include and directly re-use it.  I think we need either re-define the class in `xpu` folder, or reorg the structure of `convolution.cpp` in `at/native` which would influence all backend.